src/VIMLoader.ts

```diff
@@ -539,3 +452,108 @@ export class VIMLoader {
     )
   }
 }
+
+export class BufferGeometryBuilder {
+  defaultColor = new THREE.Color(0.5, 0.5, 0.5)
+  indexBuffer: Int32Array
+  vertexBuffer: Float32Array
+  colorBuffer: Float32Array
+
+  g3d: VimG3d
+
+  constructor (g3d: VimG3d) {
+    this.vertexBuffer = Float32Array.from(g3d.positions)
+    this.colorBuffer = new Float32Array(g3d.positions.length)
+    this.indexBuffer = new Int32Array(g3d.indices.length)
+    this.g3d = g3d
+  }
+
+  createAllGeometry = (): THREE.BufferGeometry[] => {
+    const meshCount = this.g3d.meshSubmeshes.length
+    const resultMeshes: THREE.BufferGeometry[] = []
+
+    for (let mesh = 0; mesh < meshCount; mesh++) {
+      const result = this.createBufferGeometryFromMeshIndex(mesh)
+      result?.computeBoundingSphere()
+      result?.computeBoundingBox()
```

Comment:
Are you sure you need both?

Reply:
I use sphere for centering the camera on object to provide better result for non aligned objects but maybe that's not required